### PR TITLE
Add support for testnet2

### PR DESCRIPTION
**Pull request by THenry14**
## Describe your changes

Add support for testnet2

## Linked issues

Closes https://github.com/software-mansion/starknet-jvm/issues/204

## Breaking changes

- [ ] This issue contains breaking changes

### CODE REVIEW

File: lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt [hunk 1]
```diff
@@ -439,23 +441,38 @@ class GatewayProvider(
...
     companion object Factory {
         @JvmStatic
-        fun makeTestnetClient(): GatewayProvider {
-            return GatewayProvider(
-                "$TESTNET_URL/feeder_gateway",
-                "$TESTNET_URL/gateway",
-                StarknetChainId.TESTNET,
-            )
-        }
+        fun makeTestnetClient(testnetId: StarknetChainId = StarknetChainId.TESTNET): GatewayProvider =
```
Comment (Solpatium) [inline lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt:444]: Default params don't translate well to java. Please overload methods instead.

**Review by Solpatium — CHANGES_REQUESTED** [review]

File: lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt (hunk 1 above)
Comment (THenry14) [reply lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt]: done

File: lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt [hunk 2]
```diff
@@ -19,6 +20,7 @@ import kotlinx.serialization.ExperimentalSerializationApi
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.SerializationException
 import kotlinx.serialization.json.*
+import java.lang.IllegalArgumentException
 import java.util.function.Function
```
Comment (cptartur) [inline lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt:24]: Let's not use exceptions from java.lang, but directly from kotlin.

File: lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt [hunk 3]
```diff
@@ -447,6 +449,22 @@ class GatewayProvider(
             )
         }
 
+        @JvmStatic
+        fun makeTestnetClient(testnetId: StarknetChainId): GatewayProvider =
```
Comment (cptartur) [inline lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt:453]: Should we have some kind of unit tests for these constructors?

**Review by cptartur — CHANGES_REQUESTED** [review]

File: lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt (hunk 3 above)
Comment (THenry14) [reply lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt:453]: Up until now we didn't, not sure what we could do except maybe just create a client and check its chainId (+ I'm really not sure how to test custom httpService, if at all)

File: lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt (hunk 2 above)
Comment (THenry14) [reply lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt]: Changed to `kotlin.IllegalArgumentException`

File: lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt (hunk 3 above)
Comment (cptartur) [reply lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt:453]: Perhaps let's create an issue for it then and we can cover it sometime later.

Comment (THenry14) [reply lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt:452]: https://github.com/software-mansion/starknet-jvm/issues/206

**Review by Solpatium — APPROVED** [review]

**Pull request merged**